apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -57,7 +62,110 @@ function PinButton({ unpin, onClick }: { unpin?: boolean; onClick: () => void })
 	);
 }
 
-function MyTokens({
+function TokenRowButton({
+	coinBalance,
+	page,
+	copy,
+}: {
+	coinBalance: CoinBalanceType;
+	page: 'send' | 'swap';
+	copy: string;
+}) {
+	const params = new URLSearchParams({
+		type: coinBalance.coinType,
+	});
+
+	return (
+		<ButtonOrLink
+			to={`/${page}?${params.toString()}`}
+			onClick={
+				page === 'send'
+					? () =>
+							ampli.selectedCoin({
+								coinType: coinBalance.coinType,
+								totalBalance: Number(BigInt(coinBalance.totalBalance) / MIST_PER_SUI),
+							})
+					: undefined
+			}
+			key={coinBalance.coinType}
+			className="no-underline text-subtitle font-medium text-steel hover:font-semibold hover:text-hero"
+		>
+			{copy}
+		</ButtonOrLink>
+	);
+}
+
+export function TokenRow({
+	coinBalance,
+	renderActions,
+	onClick,
+	as: Tag = 'div',
+	borderBottom,
+}: {
+	coinBalance: CoinBalanceType;
+	renderActions?: boolean;
+	as?: 'div' | 'button';
+	onClick?: () => void;
+	borderBottom?: boolean;
+}) {
+	const coinType = coinBalance.coinType;
+	const balance = BigInt(coinBalance.totalBalance);
+	const [formatted, symbol] = useFormatCoin(balance, coinType);
+	const isButton = Tag === 'button';
+
+	const isRenderSwapButton = useMemo(() => {
+		return allowedSwapCoinsList.includes(coinType);
+	}, [coinType]);
+
+	return (
+		<Tag
+			className={clsx(
+				'flex py-3 pl-1.5 pr-2 rounded hover:bg-sui/10 items-center border-solid border-t-transparent border-x-transparent bg-transparent',
+				isButton && 'hover:cursor-pointer',
+				borderBottom && 'border-b border-gray-45',
+				!borderBottom && 'border-b-transparent',
+			)}
+			onClick={onClick}
+		>
+			<div className="flex gap-2.5">
+				<CoinIcon coinType={coinType} size="md" />
+				<div className="flex flex-col gap-1 items-start">
+					<Text variant="body" color="gray-90" weight="semibold">
+						{symbol}
+					</Text>
+
+					{renderActions ? (
+						<div className="flex gap-2.5 items-center">
+							<TokenRowButton coinBalance={coinBalance} page="send" copy="Send" />
+							{isRenderSwapButton && (
+								<TokenRowButton coinBalance={coinBalance} page="swap" copy="Swap" />
+							)}
+						</div>
+					) : (
+						<div className="flex gap-1 items-center">
+							<Text variant="subtitleSmall" weight="semibold" color="gray-90">
+								{symbol}
+							</Text>
+							<Text variant="subtitleSmall" weight="medium" color="steel-dark">
+								{formatAddress(coinType)}
+							</Text>
+						</div>
+					)}
+				</div>
+			</div>
+
+			<div className="ml-auto flex flex-col items-end gap-1.5">
+				{formatted !== '0' && (
```

Comment:
I wouldn't do this check against formatted, I'd check that the bigint value is > 0. The `0` formatting should be able to change without impacting the behavior of the app.